doc/source/structures/collections/range.rst

```diff
@@ -0,0 +1,91 @@
+.. _range:
+
+Range
+=====
+
+:struct:`Range` is a type that represents a sequence of scalar whole numbers. The sequence can start and finish at any whole number, can be either descending or ascending and can skip numbers.
+
+.. note::
+
+    This is one of the few places in kOS where there is a distinction between
+    decimal (floating point, or fractional) scalar numbers and whole (integer,
+    or round) scalar numbers.  Using a decimal scalar will not throw an error,
+    however it may give unexpected results due to rounding.
+
+There are 3 ways of constructing a :struct:`Range`:
+
+- :code:`RANGE(FROM, TO, STEP)`
+  Will create a sequence of numbers that starts with `FROM` and contains all numbers `N + STEP` such that `N` belongs to the sequence and that are smaller than `TO`. For example
+  :code:`RANGE(3,8,2)` will contain numbers 3, 5 and 7. If `FROM` > `TO` then the sequence will be descending. For example :code:`RANGE(2,-9,3)` will contain numbers 2, -1, -4 and -7.
```

Comment:
I was actually going to discuss the same thing, until I started looking and found that it's the same limitation of python.  And because the default is to start at 0 this method allows you to work with zero based indices:
```
set l to list("one", "two", "three").
for i in range(l:length) {
    print l[i] + " (" + i + ")".
}
```
I'm not sure that changing the closing parameter to "end" will really convey that it is an "exclusive" limit any better than "to".  Like so many things, it's difficult to convey the mathematical meaning in a single word.  I suppose for the purpose of documentation the name of the parameter is only useful for helping the KS users understand the meaning, so we could simple change it to "from_inclusive" and "to_exclusive" (though that's implying an affinity towards "_" deliminated naming)